src/Console/Commands/EventStoreWorker.php

```diff
@@ -141,23 +128,31 @@ protected function safeGetMetadata(EventRecord $event)
 
     public function dispatch(EventRecord $eventRecord): void
     {
-        $event = $this->makeSerializableEvent($eventRecord);
+        $logger = LaravelEventStore::$logger;
+        $serializedEvent = $payload = $this->makeSerializableEvent($eventRecord);
+        $event = $serializedEvent->getType();
 
-        if ($localEvent = $this->mapToLocalEvent($event)) {
-            event($localEvent);
-        } else {
-            event($event->getType(), $event);
+        if ($localEvent = $this->mapToLocalEvent($serializedEvent)) {
+            $event = $localEvent;
+            $payload = null;
```

Comment:
I'm still not sure if functionality has not changed here and simplifying this leads to an unreadable code.

How is this easier to read than before? Please revert.

Reply:
This was pulled over from Craig's commit, would you like me to still revert?

Reply:
Yes please

Reply:
> This was pulled over from Craig's commit, would you like me to still revert?

You've changed my code! I refactored a bit as the logging is now done from this function and my logger required the type and payload as separate variables. I think I did this to match the Laravel event() method signature, but in hindsight I don't think that's required.

old
```php
    public function dispatch(EventRecord $eventRecord): void
    {
        $event = $this->makeSerializableEvent($eventRecord);

        if ($localEvent = $this->mapToLocalEvent($event)) {
            event($localEvent);
        }
        else {
            event($event->getType(), $event);
        }
    }
```


mine
```php
    public function dispatch(EventRecord $eventRecord): void
    {
        $jsonEvent = $this->makeSerializableEvent($eventRecord);
        $event = $jsonEvent->getType();
        $payload = $jsonEvent;

        if ($localEvent = $this->mapToLocalEvent($jsonEvent)) {
            $event = $localEvent;
            $payload = null;
        }

        $this->logger->eventStart($event, $payload);
        event($event, $payload);
    }
```


suggested
```php
    public function dispatch(EventRecord $eventRecord): void
    {
        $event = $this->makeSerializableEvent($eventRecord);

        $this->logger->eventStart($event);

        if ($localEvent = $this->mapToLocalEvent($event)) {
            event($localEvent);
        }
        else {
            event($event->getType(), $event);
        }
    }
```

This will require changes to the log function